### ISSUE TITLE
arrayMin/arrayMax should have a generic specialization

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#52756]
**Use case**

```
milovidov-desktop :) SELECT arrayMin([3, 1, 2])

SELECT arrayMin([3, 1, 2])

Query id: 9bf45956-442f-47a6-af83-74f40361cb66

┌─arrayMin([3, 1, 2])─┐
│                   1 │
└─────────────────────┘

1 row in set. Elapsed: 0.001 sec. 

milovidov-desktop :) SELECT arrayMin([[3], [1], [2]])

SELECT arrayMin([[3], [1], [2]])

Query id: 1d8d574a-ce78-45a0-9bba-a5b80110dbc2


0 rows in set. Elapsed: 0.001 sec. 

Received exception from server (version 23.8.1):
Code: 43. DB::Exception: Received from localhost:9000. DB::Exception: array aggregation function cannot be performed on type Array(UInt8): While processing arrayMin([[3], [1], [2]]). (ILLEGAL_TYPE_OF_ARGUMENT)

milovidov-desktop :) SELECT arrayReduce('min', [[3], [1], [2]])

SELECT arrayReduce('min', [[3], [1], [2]])

Query id: f7053fe6-4639-4a2f-85bc-acc10592d8ab

┌─arrayReduce('min', [[3], [1], [2]])─┐
│ [1]                                 │
└─────────────────────────────────────┘

1 row in set. Elapsed: 0.001 sec.
```

**Describe the solution you'd like**

`arrayMin/Max` should be applicable to all data types that have `IColumn::compareAt`.

**Comment by lohpaul9** [comment]
Hey! I can work on this with @sachinthaker

**Comment by alexey-milovidov** [comment]
@lohpaul9 thank you!

**Comment by lohpaul9** [comment]
Hey @alexey-milovidov , just to let you know - I'm still working on this but have been a little busy with things while familiarizing with the code base, will take a little bit more time

**Comment by chloro-pn** [comment]
Do we still need to implement this feature? I think I can give it a try.  
: ) @alexey-milovidov

**Comment by alexey-milovidov** [comment]
Yes, it is still relevant:

```
milovidov@milovidov-pc:~/work/ClickHouse$ ch
ClickHouse local version 24.6.1.1.

milovidov-pc :) SELECT arrayMin([[3], [1], [2]])

SELECT arrayMin([[3], [1], [2]])

Query id: 3dccf305-20be-471b-99db-d9ebe2625d97


Elapsed: 0.016 sec. 

Received exception:
Code: 43. DB::Exception: array aggregation function cannot be performed on type Array(UInt8): In scope SELECT arrayMin([[3], [1], [2]]). (ILLEGAL_TYPE_OF_ARGUMENT)

milovidov-pc :)
```

Let's try!

**Comment by rschu1ze** [comment]
Resolved by #65455
